src/Kestrel.Core/KestrelServerOptions.cs

```diff
@@ -103,6 +103,7 @@ public void ConfigureEndpointDefaults(Action<ListenOptions> configureOptions)
         internal void ApplyEndpointDefaults(ListenOptions listenOptions)
         {
             listenOptions.KestrelServerOptions = this;
+            ConfigurationLoader?.ApplyConfigurationDefaults(listenOptions);
```

Comment:
When is ConfigurationLoader null?

Reply:
It's null by default. It's only set by CreateDefaultBuilder:
https://github.com/aspnet/MetaPackages/blob/5605eabee6be4790f2fd22b27a2eb9c92070be23/src/Microsoft.AspNetCore/WebHost.cs#L163